### ISSUE TITLE
docs [import-style]: Improve documentation of import-style

### ISSUE THREAD

**Issue by sharky98** [sindresorhus/eslint-plugin-unicorn#2154]
### A - Application of the rule

By simply reading the [docs] it isn't clear if that rule applies only to the import of the three listed modules (util, path, chalk) or all modules when using the default `"plugin:unicorn/recommended"`.

My first assumption was that style `default` was prioritized over the other, while in fact it's only the three modules shown that are affected.

[docs]:https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/import-style.md

#### Examples

* ❌ Not triggered on import `import * as React from "react";`.
* ✅ Triggered on import `import * as path from "node:path";`.

#### Proposed modification

```diff
Sometimes a module contains unrelated functions, like `util`, thus it is a good practice to enforce destructuring or named imports here. Other times, in modules like `path`, it is good to use default import as they have similar functions, all likely to be utilized.

+This rule applies only to specified modules in `styles` option. If a module is not listed in `styles`, it will not be linted.

This rule defines 4 import styles:
```

```diff
### styles

Type: `object`

-You can extend default import styles per module by passing the `styles` option.
+You can extend the import styles accepted for each module individually by passing the `styles` option and adding a key for each module.

Default options per module are:
```

### B - Contradiction with default `tsconfig.json`

Since default configuration for `tsconfig.json` are `false` for both [`allowSyntheticDefaultImports`][tsconfig-01] and [`esModuleInterop`][tsconfig-02], using default import is not authorized; it raise error TS1259.

[tsconfig-01]:https://www.typescriptlang.org/tsconfig#allowSyntheticDefaultImports

[tsconfig-02]:https://www.typescriptlang.org/tsconfig#esModuleInterop

#### Proposed modification

```diff
Sometimes a module contains unrelated functions, like `util`, thus it is a good practice to enforce destructuring or named imports here. Other times, in modules like `path`, it is good to use default import as they have similar functions, all likely to be utilized.

+💫 Using the default configuration of this rule will contradict with configuration of TypeScript `allowSyntheticDefaultImports = false` and `esModuleInterop = false`. In such case, you can disable this rule or extend the styles to allow `namespace` import style for the module that requires it like `path` or `react`.

This rule defines 4 import styles:
```

Feel free to use another emoji, I used the dizzy one, since I was so dizzy or figuring this out 😅!

**Comment by sindresorhus** [comment]
> Since default configuration for tsconfig.json are false for both [allowSyntheticDefaultImports](https://www.typescriptlang.org/tsconfig#allowSyntheticDefaultImports) and [esModuleInterop](https://www.typescriptlang.org/tsconfig#esModuleInterop), using default import is not authorized; it raise error TS1259.

`esModuleInterop` is enabled by default `module` is `node16` or `nodenext`, which is expected config for using ESM and `import`.

**Comment by sharky98** [comment]
While this is true; and `esModuleInterop` is recommended (most [bases](https://github.com/tsconfig/bases/) set it true against the default), it does come with issues for library makers. See the following discussions.

- https://github.com/microsoft/TypeScript/issues/52779
- https://github.com/microsoft/TypeScript/issues/41139#issuecomment-1431499324
- https://github.com/plantain-00/type-coverage/issues/78

My understanding reading these is that for an end-user project, `esModuleInterop` should be set to true. However, for a module distributed, it should be set to false.

- https://evertpot.com/universal-commonjs-esm-typescript-packages/#esmoduleinterop
